lib/zebra/print_job.rb

```diff
@@ -16,15 +16,22 @@ def print(label, ip)
       else
         tempfile = label.persist
       end
-      send_to_printer tempfile.path
+      send_to_printer(tempfile.path, print_service)
     end
 
     private
 
-    def send_to_printer(path)
+    def send_to_printer(path, print_service)
+      case print_service
+      when "rlpr"
+        system("rlpr -H #{@remote_ip} -P #{@printer} -o #{path} 2>&1") # try printing to LPD on windows machine 
+      when "lp"
+        system("lp -h #{@remote_ip} -d #{@printer} -o raw #{path}") # print to unix (CUPS) using lp
+      else 
+        result = system("rlpr -H #{@remote_ip} -P #{@printer} -o #{path} 2>&1") # try printing to LPD on windows machine first
+        system("lp -h #{@remote_ip} -d #{@printer} -o raw #{path}") if !result # print to unix (CUPS) if rlpr failed
+      end
       puts "* * * * * * * * * * * * Sending file to printer #{@printer} at #{@remote_ip} * * * * * * * * * * "
```

Comment:
I'm going to push this line back up to the start of the method, in case anyone was relying on the return value.